tests/ga/test_extension_telemetry.py

```diff
@@ -0,0 +1,539 @@
+            if match is not None:
+                expected_handler_name = match.group("name")
+                self.assertIn(expected_handler_name, handler_name_with_count, "Extension event not found")
+                handler_name_with_count.pop(expected_handler_name)
+
+        self.assertEqual(len(handler_name_with_count), 0, "All extension events not matched")
+
+    def test_it_should_trim_message_if_more_than_limit(self):
+        max_len = 100
+        no_of_extensions = 2
+        with patch("azurelinuxagent.ga.extension_telemetry.ProcessExtensionTelemetry._EXTENSION_EVENT_MAX_MSG_LEN", max_len):
+            handler_name_with_count, event_body = self._setup_and_assert_tests_for_max_sizes()
+            context1_vals = self._get_param_value_from_event_body_if_exists(event_body,
+                                                                            GuestAgentGenericLogsSchema.Context1)
+            self.assertEqual(no_of_extensions, len(context1_vals),
+                             "There should be {0} Context1 values".format(no_of_extensions))
+
+            for val in context1_vals:
+                self.assertLessEqual(len(val), max_len, "Message Length does not match")
+
+    def test_it_should_skip_events_larger_than_max_size_and_report_event(self):
+        max_size = 1000
+        no_of_extensions = 3
+        with patch("azurelinuxagent.ga.extension_telemetry.add_log_event") as mock_event:
+            with patch("azurelinuxagent.ga.extension_telemetry.ProcessExtensionTelemetry._EXTENSION_EVENT_MAX_SIZE",
+                       max_size):
+                handler_name_with_count, _ = self._setup_and_assert_tests_for_max_sizes(no_of_extensions, expected_count=0)
+                self._assert_invalid_extension_error_event_reported(mock_event, handler_name_with_count,
+                                                                    error=InvalidExtensionEventError.OversizeEventError)
+
+    def test_it_should_skip_large_files_greater_than_max_file_size_and_report_event(self):
+        max_file_size = 10000
+        no_of_extensions = 5
+        with patch("azurelinuxagent.ga.extension_telemetry.add_log_event") as mock_event:
+            with patch("azurelinuxagent.ga.extension_telemetry.ProcessExtensionTelemetry._EXTENSION_EVENT_FILE_MAX_SIZE",
+                       max_file_size):
+                handler_name_with_count, _ = self._setup_and_assert_tests_for_max_sizes(no_of_extensions, expected_count=0)
+
+                pattern = r'Skipping file:\s*{0}/(?P<name>.+?)/{1}.+'.format(conf.get_ext_log_dir(), EVENTS_DIRECTORY)
+                self._assert_event_reported(mock_event, handler_name_with_count, pattern)
+
+    def test_it_should_map_extension_event_json_correctly_to_telemetry_event(self):
+
+        # EventName maps to HandlerName + '-' + Version from event file
+        expected_mapping = {
+            GuestAgentGenericLogsSchema.EventName: ExtensionEventSchema.Version,
```

Comment:
Confused between this line and the comment above, the definitions don't seem to match?

Reply:
Ohh yeah sorry for the confusion, in the code below I'm actually mapping it to "Name-Version". This dict is supposed to just be a placeholder

Reply:
Its happening in Line 433 of this file